modules/channel-web/src/views/lite/store/index.ts

```diff
@@ -428,14 +429,28 @@ class RootStore {
 
     document.title = this.config.botName || 'Botpress Webchat'
 
+    this.api.updateAxiosConfig({ botId: this.config.botId, externalAuthToken: this.config.externalAuthToken })
+    this.api.updateUserId(this.config.userId)
+
     try {
-      window.USE_SESSION_STORAGE = this.config.useSessionStorage
-    } catch {
-      console.error('Could not set USE_SESSION_STORAGE')
+      if (window.USE_SESSION_STORAGE !== this.config.useSessionStorage) {
+        window.USE_SESSION_STORAGE = this.config.useSessionStorage
+        // Reconfigure the EventBus since the storage provider has changed
+        this.bp.events.setup()
+
+        await waitForUserId()
```

Comment:
why do we want to get the user Id here ? isn't it already set ?

Reply:
Calling setup will **re-create** a new socket connection but using the session storage instead of the local storage. We have to wait for the userId here also, for the same reason we have to wait on the first setup.